### PR TITLE
Update CHANGELOG.md with v3.8.0 changes

### PR DESCRIPTION
**Pull request by dataclouder**
* Update CHANGELOG.md
* UPDATE Go version in README.md
* Fix headers in CHANGELOG.md and update script to produce them correctly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,115 @@
...
+## BUG FIXES
+* Fix bug where VM was power cycled multiple times during creation ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+* Fix bug where storage_profile is ignored for empty (non template) VM ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+* `resource/vcd_nsxt_alb_edgegateway_service_engine_group` field `reserved_virtual_services` accepts
+  "0" as value ([#923](https://github.com/vmware/terraform-provider-vcd/pull/923))
+* Fixed a bug in `resource/vcd_vapp` that would prevent to Power off vApp when previous state was
+  `power_on=true` ([#932](https://github.com/vmware/terraform-provider-vcd/pull/932))
+
+## DEPRECATIONS
+Deprecate `vcd_external_network` in favor of `vcd_external_network_v2` ([#903](https://github.com/vmware/terraform-provider-vcd/pull/903))
```
Comment (adambarreiro) [inline CHANGELOG.md:73]:
```suggestion
* Deprecate `vcd_external_network` in favor of `vcd_external_network_v2` ([#903](https://github.com/vmware/terraform-provider-vcd/pull/903))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,115 @@
...
+* Deprecated `default_vm_sizing_policy_id` field in `vcd_org_vdc` resource and data source. This field is misleading as it
```
Comment (adambarreiro) [inline CHANGELOG.md:74]: As a general comment, probably we should align all verb tenses.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,115 @@
...
+  `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`, `vcd_org_vdc`, `vcd_provider_vdc`,
+  `vcd_storage_profile`, `vcd_vapp`, `vcd_vapp_vm`. This new attribute replaces `metadata`
+  to add support of metadata visibility (user access levels), all the available types and domains for every metadata
+  entry. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
+* Add attribute `metadata_entry` to the following resources:
+  `vcd_catalog`, `vcd_catalog_media`, `vcd_independent_disk`, `vcd_network_direct`, `vcd_network_isolated`,
+  `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`, `vcd_org_vdc`, `vcd_vapp`,
+  `vcd_vapp_vm`. This new attribute replaces `metadata` to add support of metadata visibility (user access levels),
+  all the available types and domains for every metadata entry. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
+* Add `placement_policy_id` attribute to `vcd_vapp_vm` and `vcd_vm` resource and data source,
```
Comment (adambarreiro) [inline CHANGELOG.md:27]: Sometimes the barrier between Feature and Improvement is fuzzy to me, I noticed I placed this inside Features, but  there's
```
Add `vm_placement_policy_ids` attribute to `vcd_org_vdc`
```
in Improvements section.
Should we move this one into Improvements?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,115 @@
...
+  `status_text` ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+* Add `vm_placement_policy_ids` attribute to `vcd_org_vdc` resource and data source to assign existing
+  VM Placement Policies to VDCs ([#904](https://github.com/vmware/terraform-provider-vcd/pull/904), [#911](https://github.com/vmware/terraform-provider-vcd/pull/911))
+* Add `default_compute_policy_id` attribute to `vcd_org_vdc` resource and data source to specify a default
+  VM Sizing Policy, VM Placement Policy or vGPU Policy for the VDC ([#904](https://github.com/vmware/terraform-provider-vcd/pull/904), [#911](https://github.com/vmware/terraform-provider-vcd/pull/911))
+* Add attributes `href`, `vapp_template_list`, `media_item_list`, and `publishing_url` to `vcd_catalog` resource and data source to show published items ([#916](https://github.com/vmware/terraform-provider-vcd/pull/916))
+* Add `subscribed_catalog` to examples ([#916](https://github.com/vmware/terraform-provider-vcd/pull/916))
+* Upgrade Terraform SDK dependency to v2.24.1 ([#920](https://github.com/vmware/terraform-provider-vcd/pull/920), [#930](https://github.com/vmware/terraform-provider-vcd/pull/930))
```
Comment (adambarreiro) [inline CHANGELOG.md:39]: Improvement or note?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,115 @@
...
+* Data source `vcd_nsxt_edge_cluster` deprecated `vdc` field in favor of 3 new fields to define
```
Comment (adambarreiro) [inline CHANGELOG.md:40]: Should this be rephrased? Or moved to deprecations?

**Review by adambarreiro — COMMENTED** [review]
First scan

File: CHANGELOG.md (hunk 2 above)
Comment (dataclouder) [reply CHANGELOG.md]: Done

**Review by adambarreiro — APPROVED** [review]
LGTM!

**Review by Didainius — APPROVED** [review]

**Comment by adambarreiro** [comment]
Just a doubt, should we fix individual changelog files like in https://github.com/vmware/go-vcloud-director/pull/523?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,10 +1,120 @@
...
+* **New Data Source**: `vcd_subscribed_catalog` that allows reading a subscribed catalog ([#916](https://github.com/vmware/terraform-provider-vcd/pull/916))
+* **New Data Source**: `vcd_task` that allows reading a VCD task ([#916](https://github.com/vmware/terraform-provider-vcd/pull/916))
+
+### IMPROVEMENTS
+* Add attribute `metadata_entry` to the following data sources:
+  `vcd_catalog`, `vcd_catalog_media`, `vcd_independent_disk`, `vcd_network_direct`, `vcd_network_isolated`,
+  `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`, `vcd_org_vdc`, `vcd_provider_vdc`,
+  `vcd_storage_profile`, `vcd_vapp`, `vcd_vapp_vm`. This new attribute replaces `metadata`
+  to add support of metadata visibility (user access levels), all the available types and domains for every metadata
+  entry. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```
Comment (lvirbalas) [inline CHANGELOG.md:23]:
```suggestion
  entry ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,10 +1,120 @@
...
+* Add attribute `metadata_entry` to the following resources:
+  `vcd_catalog`, `vcd_catalog_media`, `vcd_independent_disk`, `vcd_network_direct`, `vcd_network_isolated`,
+  `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`, `vcd_org_vdc`, `vcd_vapp`,
+  `vcd_vapp_vm`. This new attribute replaces `metadata` to add support of metadata visibility (user access levels),
+  all the available types and domains for every metadata entry. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```
Comment (lvirbalas) [inline CHANGELOG.md:28]:
```suggestion
  all the available types and domains for every metadata entry ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,10 +1,120 @@
...
+  * `vcd_vapp_access_control`
+  * `vcd_nsxt_alb_edgegateway_service_engine_group`
+  * `vcd_org_vdc`
+  * `vcd_org_user`
+  * `vcd_external_network`
+* Resource and data source `vcd_nsxt_network_dhcp` support Isolated networks, different DHCP modes
+  ('EDGE', 'NETWORK', 'RELAY') and lease time ([#929](https://github.com/vmware/terraform-provider-vcd/pull/929))
+* Add the new attributes `vapp_template_id`, `boot_image_id` to the resources `vcd_vapp_vm` and `vcd_vm` to be able
+  to use unique URNs to reference vApp Templates and Media items through data sources, to build strong dependencies
+  in Terraform configuration. ([#931](https://github.com/vmware/terraform-provider-vcd/pull/931))
```
Comment (lvirbalas) [inline CHANGELOG.md:60]:
```suggestion
  in Terraform configuration ([#931](https://github.com/vmware/terraform-provider-vcd/pull/931))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,10 +1,120 @@
...
+
+### BUG FIXES
+* Fix bug where VM was power cycled multiple times during creation ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+* Fix bug where storage_profile is ignored for empty (non template) VM ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+* `resource/vcd_nsxt_alb_edgegateway_service_engine_group` field `reserved_virtual_services` accepts
+  "0" as value ([#923](https://github.com/vmware/terraform-provider-vcd/pull/923))
+* Fixed a bug in `resource/vcd_vapp` that would prevent to Power off vApp when previous state was
```
Comment (lvirbalas) [inline CHANGELOG.md:67]:
```suggestion
* Fix a bug in `resource/vcd_vapp` that prevented Power Off vApp when previous state was
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,10 +1,120 @@
...
+
+### DEPRECATIONS
+* Deprecate `vcd_external_network` in favor of `vcd_external_network_v2` ([#903](https://github.com/vmware/terraform-provider-vcd/pull/903))
+* Deprecate `default_vm_sizing_policy_id` field in `vcd_org_vdc` resource and data source. This field is misleading as it
+  can contain not only VM Sizing Policies but also VM Placement Policies or vGPU Policies.
+  Its replacement is the `default_compute_policy_id` attribute ([#904](https://github.com/vmware/terraform-provider-vcd/pull/904))
+* Deprecate attribute `metadata` in favor of `metadata_entry` in the following data sources:
+  `vcd_catalog`, `vcd_catalog_media`, `vcd_catalog_vapp_template`, `vcd_independent_disk`, `vcd_network_direct`,
+  `vcd_network_isolated`, `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`,
+  `vcd_org_vdc`, `vcd_provider_vdc`, `vcd_storage_profile`, `vcd_vapp`, `vcd_vapp_vm`. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```
Comment (lvirbalas) [inline CHANGELOG.md:78]:
```suggestion
  `vcd_org_vdc`, `vcd_provider_vdc`, `vcd_storage_profile`, `vcd_vapp`, `vcd_vapp_vm` ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,10 +1,120 @@
...
+* Deprecate attribute `metadata` in favor of `metadata_entry` in the following resources:
+  `vcd_catalog`, `vcd_catalog_media`, `vcd_catalog_vapp_template`, `vcd_independent_disk`, `vcd_network_direct`,
+  `vcd_network_isolated`, `vcd_network_isolated_v2`, `vcd_network_routed`, `vcd_network_routed_v2`, `vcd_org`,
+  `vcd_org_vdc`, `vcd_vapp`, `vcd_vapp_vm`. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```
Comment (lvirbalas) [inline CHANGELOG.md:82]:
```suggestion
  `vcd_org_vdc`, `vcd_vapp`, `vcd_vapp_vm` ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,10 +1,120 @@
...
+* Deprecate attribute `catalog_item_metadata` in favor of `metadata_entry` in the `vcd_catalog_item` resource
+  and data source. ([#917](https://github.com/vmware/terraform-provider-vcd/pull/917))
+* Deprecate `template_name` in favor of `vapp_template_id` in `vcd_vapp_vm` and `vcd_vm` to be able to use unique URNs instead
+  of catalog dependent names ([#931](https://github.com/vmware/terraform-provider-vcd/pull/931))
+* Deprecate `boot_image` in favor of `boot_image_id` in `vcd_vapp_vm` and `vcd_vm` to be able to use URNs instead
+  of catalog dependent names ([#931](https://github.com/vmware/terraform-provider-vcd/pull/931))
+* Deprecate `catalog_name` in favor of `vapp_template_id` or `boot_image_id`, which don't require a catalog name anymore ([#931](https://github.com/vmware/terraform-provider-vcd/pull/931))
+* Data source `vcd_nsxt_edge_cluster` deprecates `vdc` field in favor of 3 new fields to define
```
Comment (lvirbalas) [inline CHANGELOG.md:90]:
```suggestion
* Data source `vcd_nsxt_edge_cluster` deprecates `vdc` field in favor of three new fields to define
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,10 +1,120 @@
...
+  NSX-T Edge Cluster lookup scope - `vdc_id`, `vdc_group_id`, and `provider_vdc_id` ([#921](https://github.com/vmware/terraform-provider-vcd/pull/921))
```
Comment (lvirbalas) [inline CHANGELOG.md:91]: I can't seem to find a changelog entry for these three new fields? If they are new, they should be listed not only in deprecations section.

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,10 +1,120 @@
...
+  listed below ([#901](https://github.com/vmware/terraform-provider-vcd/pull/901))
+  * `vcd_vapp_vm` built from vApp template
+  * `vcd_vm` built from vApp template
+  * `vcd_vapp_vm` built without vApp template (empty VM)
+  * `vcd_vm` built without vApp template (empty VM)
+* Bump Go to 1.19 in `go.mod` as the minimum required version. ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902), [#916](https://github.com/vmware/terraform-provider-vcd/pull/916))
+* Code documentation formatting is adjusted using Go 1.19 (`make fmt`) ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902))
+* Adjust GitHub actions in pipeline to use the latest code ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902))
+* `staticcheck` switched version naming from `2021.1.2` to `v0.3.3` in downloads section. This PR
+  also updates the code to fetch correct staticcheck. ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902))
```
Comment (lvirbalas) [inline CHANGELOG.md:108]:
```suggestion
  also updates the code to fetch correct staticcheck ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902))
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,10 +1,120 @@
...
+* package `io/ioutil` is deprecated as of Go 1.16. `staticcheck` started complaining about usage of
+  deprecated packages. As a result this PR switches packages to either `io` or `os` (still the same
+  functions are used) ([#902](https://github.com/vmware/terraform-provider-vcd/pull/902))
+* Drop support for EOL VCD 10.2.x ([#903](https://github.com/vmware/terraform-provider-vcd/pull/903))
```
Comment (lvirbalas) [inline CHANGELOG.md]: Please move this bullet point to the top of the list (as it may be one of the most important for some users).

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 13 above)
Comment (dataclouder) [reply CHANGELOG.md:91]: This line was in "improvements" but I was told to move it to deprecations

File: CHANGELOG.md (hunk 15 above)
Comment (dataclouder) [reply CHANGELOG.md]: It is already at the top of the list. Please refresh

**Review by lvirbalas — APPROVED** [review]
Let's just have in mind that the source .md files are a bit outdated as of now :)

**Pull request merged**